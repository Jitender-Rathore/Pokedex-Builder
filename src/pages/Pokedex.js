import React from "react";
import { Link } from "react-router-dom";
import PokeCardsCreator from "../components/PokeCardsCreator";
import pokemonBall from "../images/pokemon_ball.png";

function Pokedex(props) {

const { pokemons, setPokemons } = props;

//state set for search ,searchresults , searchLoader , setlanguage 
const [searchTerm, setSearchTerm] = React.useState("");
const [searchResults, setSearchResults] = React.useState([]);
const [searchLoader, setSearchLoader] = React.useState(false);
const [language, setLanguage] = React.useState("english");


//handle change for search pokemons
const handleSearchChange = (event) => {
  setSearchTerm(event.target.value);
};

//handle change on language selection and setting the value
const handleLangChange = (event) => {
  setLanguage(event.target.value);
};

/*
setting the search loader to true and search the pokemons
  acc to user search input and  if the search term is  empty 
  set the results to pokemon list passed as props and then set
  the searchloader to false
*/
React.useEffect(() => {
  setSearchLoader(true);
  const timeoutID = setTimeout(function () {
    const results = searchTerm
      ? pokemons.filter((pokemon) =>
          pokemon.name.english.toLowerCase().includes(searchTerm.toLowerCase())
        )
      : pokemons;
    setSearchResults(results);
    setSearchLoader(false);
  }, 200);

  return () => clearTimeout(timeoutID);
}, [searchTerm, pokemons]);


/* 
    remove particular pokemon and updating the setpokemons
    Takes id of the pokemon as an input , remove it from the list 
    and return the new list of pokemons
*/
function removePokemon(id) {
  const newPokemons = pokemons.filter((pokemon) => {
    return pokemon.id !== id;
  });
  setPokemons(newPokemons);
}


return (
  <>
    <div className="header lang">

        {/*Language Change Options*/ }
      <select value={language} onChange={handleLangChange}>
        <option value="english">English</option>
        <option value="japanese">Japanese</option>
        <option value="chinese">Chinese</option>
      </select>
        
        {/*Add Button Link*/ }
      <Link to="/add">
        <div className="addPokemonWrapper">
          <div className="addPokeImage">
            <img className="pokeballImage" alt="pokeBall" src={pokemonBall} />
            <h3> Add Pokemon </h3>
          </div>
        </div>
      </Link>

    </div>

    {/*Search Bar*/ }
    <div className="searchBoxWrapper">
      <div className={`searchBox ${searchLoader ? "loading" : ""}`}>
        <input
          type="text"
          placeholder="Search"
          value={searchTerm}
          onChange={handleSearchChange}
        />
      </div>
    </div>

    {/*PokemonCardCreation by mapping searchresults*/ }
    <div className="App">
      {searchResults.map((pokemon) => {
        return (
          <PokeCardsCreator
            pokemon={pokemon}
            language={language}
            removePokemon={removePokemon}
            key={pokemon.id}
          />
        );
      })}
    </div>

  </>
);

}

export default Pokedex;
