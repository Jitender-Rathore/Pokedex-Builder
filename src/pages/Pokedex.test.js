import {shallow} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Pokedex from './Pokedex';
import {BrowserRouter} from 'react-router-dom'

describe('<Pokedex />', () => {
  const shallowComponent = () => shallow(<Pokedex />);

  it('renders correctly', () => {
    const tree = renderer.create( <BrowserRouter><Pokedex /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  /*it('should render copyright on the license', () => {
    const wrapper = shallowComponent();
    expect(wrapper.find('.license').text()).toContain(
      'Copyright (c) 2018 Martin Garcia Monterde',
    );
  });*/
});