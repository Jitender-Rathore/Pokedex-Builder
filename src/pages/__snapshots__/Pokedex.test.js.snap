// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pokedex /> renders correctly 1`] = `
Array [
  <div
    className="header lang"
  >
    <select
      onChange={[Function]}
      value="english"
    >
      <option
        value="english"
      >
        English
      </option>
      <option
        value="japanese"
      >
        Japanese
      </option>
      <option
        value="chinese"
      >
        Chinese
      </option>
    </select>
    <a
      href="/add"
      onClick={[Function]}
    >
      <div
        className="addPokemonWrapper"
      >
        <div
          className="addPokeImage"
        >
          <img
            alt="pokeBall"
            className="pokeballImage"
            src="pokemon_ball.png"
          />
          <h3>
             Add Pokemon 
          </h3>
        </div>
      </div>
    </a>
  </div>,
  <div
    className="searchBoxWrapper"
  >
    <div
      className="searchBox "
    >
      <input
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
  </div>,
  <div
    className="App"
  />,
]
`;
